wayland-scanner/src/common_gen.rs

```diff
@@ -626,11 +626,50 @@ pub(crate) fn gen_interface(
     low_name: &str,
     version: u32,
     addon: Option<TokenStream>,
+    side: Side,
 ) -> TokenStream {
+    let object_type = Ident::new(
+        match side {
+            Side::Client => "Proxy",
+            Side::Server => "Resource",
+        },
+        Span::call_site(),
+    );
```

Comment:
We already have functionality for that: the `object_name()` method of `Side` returns an `Ident` with the proper value.